src/abstractdataframe/abstractdataframe.jl

```diff
@@ -373,6 +373,18 @@ function Base.isequal(df1::AbstractDataFrame, df2::AbstractDataFrame)
     return true
 end
 
+"""
+    isapprox(df1::AbstractDataFrame, df2::AbstractDataFrame)
+
+    Test for "approximate" equality between two DataFrames
+"""
+
+function Base.isapprox(df1::AbstractDataFrame, df2::AbstractDataFrame)
+    size(df1, 2) == size(df2, 2) || return false
+    isequal(index(df1), index(df2)) || return false
+    Matrix(df1) ≈ Matrix(df2) || return false
```

Comment:
the whole point of this PR is not do do conversion to `Matrix`.

Reply:
Can you suggest an alternative to matrix conversion ? 
Should I align with the `isequal` approach?

Reply:
yes. But it order to stay consistent with Base the `norm` should be calculated using the rules that are defined by LinearAlgebra/src/generic.jl:1588.

Reply:
also probably it will be more complex than for `isequal` as we also have to handle `missing` efficiently which means that probably a barrier function would be useful there.